httpbinding/user_agent.go

```diff
@@ -0,0 +1,38 @@
+package httpbinding
+
+import (
+	"fmt"
+	"strings"
+)
+
+// UserAgentBuilder is a builder for a HTTP User-Agent string.
+type UserAgentBuilder struct {
+	sb *strings.Builder
+}
+
+// NewUserAgentBuilder returns a new UserAgentBuilder.
+func NewUserAgentBuilder() *UserAgentBuilder {
+	return &UserAgentBuilder{sb: &strings.Builder{}}
+}
```

Comment:
not too important but don't really need to take a pointer for `strings.Builder` since its a member of a struct that is a pointer.